packages/openapi-v3-types/src/openapi-v3-spec-types.ts

```diff
@@ -45,336 +50,6 @@ export type ParameterStyle =
-  description?: string;
-  format?: string;
-  default?: ExtensionValue;
-}
-
-/**
- * JSON type - This is part of the Schema object.
- * The following properties are taken directly from the JSON Schema
- * definition and follow the same specifications.
- * See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schema-object
- */
-
-export type JSONType = {
-  title?: string;
-  multipleOf?: number;
-  maximum?: number;
-  exclusiveMaximum?: number;
-  minimum?: number;
-  exclusiveMinimum?: number;
-  maxLength?: number;
-  minLength?: number;
-  // (This string SHOULD be a valid regular expression, according to the ECMA 262 regular expression dialect)
-  pattern?: string;
-  maxItems?: number;
-  minItems?: number;
-  uniqueItems?: boolean;
-  maxProperties?: number;
-  minProperties?: number;
-  enum?: Array<ExtensionValue>;
-};
-
-/**
- * Describes a single request body.
- * Specification:
- * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#request-body-object
- */
-export interface RequestBodyObject extends ISpecificationExtension {
-  description?: string;
-  content: ContentObject;
-  required?: boolean;
-}
-
-/**
- * Describes an object of multiple content types.
- * For example:
- * ```js
- * {
- *   'application/json': {
- *     schema: {...schemaObjectSpec}
- *   },
- *   'application/text': {
- *     schema: {...schemaObjectSpec}
- *   }
- * }
- * ```
- */
-export interface ContentObject {
-  [mediatype: string]: MediaTypeObject;
-}
-
-/**
- * Each Media Type Object provides schema and examples for the media type
- * identified by its key.
- * Specification:
- * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#media-type-object
- */
-export interface MediaTypeObject extends ISpecificationExtension {
-  schema?: SchemaObject | ReferenceObject;
-  examples?: [ExampleObject | ReferenceObject];
-  example?: ExampleObject | ReferenceObject;
-  encoding?: EncodingObject;
-}
-
-/**
- * Describes an encoding object, used in ParameterObject
- */
-export interface EncodingObject extends ISpecificationExtension {
-  [property: string]: EncodingPropertyObject | ExtensionValue;
-}
-
-/**
- * Describes an encoding object, used in `SchemaObject`
- */
-export interface EncodingPropertyObject {
-  contentType?: string;
-  headers?: {[key: string]: HeaderObject | ReferenceObject};
-  style?: string;
-  explode?: boolean;
-  allowReserved?: boolean;
-  [key: string]: ExtensionValue;
-}
-
-/**
- * Describes a header object, used in `EncodingPropertyObject`
- */
-export interface HeaderObject extends ParameterObject {}
-
-/**
- * Describes a single operation parameter.
- * A unique parameter is defined by a combination of a name and location.
- * Specification
- * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameter-object
- */
-export interface ParameterObject extends ISpecificationExtension {
-  name: string;
-  in: ParameterLocation;
```

Comment:
@raymondfeng The `openapi3-ts` package doesn't have `ParameterLocation` and `ParameterStyle`, so its `ParameterObject` doesn't verify the value of property `in` and `style` are valid